docs/source/services/timelock_service/migration.rst

```diff
@@ -0,0 +1,145 @@
+.. _timelock-migration:
+
+Migration to External Timelock Services
+=======================================
+
+.. danger::
+
+   Improperly configuring one's cluster to use external timestamp and lock services can result in **SEVERE DATA
+   CORRUPTION**! Please contact the AtlasDB team before attempting a migration.
+
+Why Migration?
+--------------
+
+AtlasDB assumes that timestamps returned by the timestamp service are monotonically increasing. In order to preserve
+this guarantee when moving from an embedded timestamp service to an external timestamp service, we need to ensure
+that timestamps issued by the external timestamp service are larger than those issued by the embedded one.
+Otherwise, this can lead to serious data corruption.
+
+The migration process must be run offline (that is, with no AtlasDB clients running during migration) and basically
+consists of the following steps:
+
+1. Set up the external timestamp service.
+2. Obtain a timestamp ``TS`` that is at least as large as the largest timestamp issued.
+3. Fast-forward the external timestamp service to at least ``TS``.
+4. Invalidate the old timestamp service, preventing AtlasDB clients from retrieving timestamps from it.
+5. Configure AtlasDB clients to retrieve timestamps from the new timestamp service.
+6. Restart AtlasDB clients.
+
+Strictly speaking, step 4 is not required; that said, we prefer to do this to avoid the risk of data corruption in the
+event of misconfiguration (that is, if some but not all clients are pointed to the new timestamp service).
+
+Manual Migration
```

Comment:
Great manual steps! I didn't actually expect us to document this at all given we're automating it, but I think having the logic here is very clear and we can share exactly how the process works with this document alone.